src/components/Rockets.js

```diff
@@ -1,39 +1,63 @@
 import { useDispatch, useSelector } from 'react-redux';
 import { useEffect } from 'react';
-import { getRockets } from '../redux/rocketsSlice';
-// import '../styles/Rockets.css';
+import {
+  Button, Container, Row, Col,
+} from 'react-bootstrap';
+import { getRockets, reserveRocket, cancelRocket } from '../redux/rocketsSlice';
+import '../styles/rockets.css';
 
 // GET ROCKETS DATA FROM REDUX STORE
 const Rockets = () => {
   const dispatch = useDispatch();
   const rockets = useSelector((state) => state.rockets.rockets);
-  // console.log('Rockets:', rockets);
 
   // FETCH ROCKETS DATA ON PAGE LOAD
   useEffect(() => {
     dispatch(getRockets());
   }, [dispatch]);
 
+  // HANDLE RESERVE/CANCEL ROCKET
+  const handleReserveRocket = (id, reserved) => {
+    if (reserved) {
+      dispatch(cancelRocket(id));
+    } else {
+      dispatch(reserveRocket(id));
+    }
+  };
```

Comment:
- [ ] Same thing for the Rockets, Whenever the Rockets component is rendered , the `state` of all rockets  is reset 😄 . Which cause the Rockets the used has reserved to be reset. To fix that , you should dispatch the  `getRockets` in the `App.js` because it is called once 👍 . Kindly update that


```jsx
...
import { useEffect } from 'react'; 
import { getMissions } from './redux/missionsSlice';
import { getRockets } from './redux/rocketsSlice';
...

function App() {
  const dispatch = useDispatch();
    useEffect(() => {
    dispatch(getRockets());
    dispatch(getMissions());
  }, [dispatch]);
...
}
```